scala-package/spark/src/test/scala/org/apache/mxnet/spark/MXNetGeneralSuite.scala

```diff
@@ -42,30 +44,38 @@ class MXNetGeneralSuite extends SharedSparkContext {
   }
 
   private def downloadTestData(): Unit = {
+    logger.info("Downloading the training data...")
     Process("wget http://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon" +
       "/dataset/mxnet-spark-test/train.txt" + " -P " + testDataDir + " -q") !
   }
 
-//  override def beforeAll(): Unit = {
-//  val tempDirFile = Files.createTempDirectory(s"mxnet-spark-test-${System.currentTimeMillis()}").
-//      toFile
-//    testDataDir = tempDirFile.getPath
-//    tempDirFile.deleteOnExit()
-//    downloadTestData()
-//  }
+  override def beforeAll(): Unit = {
+  val tempDirFile = Files.createTempDirectory(s"mxnet-spark-test-${System.currentTimeMillis()}").
+      toFile
+    testDataDir = tempDirFile.getPath
+    tempDirFile.deleteOnExit()
+    downloadTestData()
+  }
 
-  test("Dummy test on Spark") {
+  test("run spark with MLP") {
+    if (System.getenv().containsKey("SCALA_TEST_ON_GPU") &&
+      System.getenv("SCALA_TEST_ON_GPU").toInt == 1) {
+      val trainData = parseRawData(sc, s"$testDataDir/train.txt")
+      val model = buildMlp().fit(trainData)
+      assert(model != null)
+    } else {
+      logger.info("Currently not supporting CPU, skipped for now")
+    }
+  }
 
+  test("run spark with LeNet") {
+    if (System.getenv().containsKey("SCALA_TEST_ON_GPU") &&
```

Comment:
it consistently pass in GPU but not CPU??????? seems surprise to me

Reply:
Nope, GPU do have an issue

Reply:
https://github.com/apache/incubator-mxnet/issues/11263

Reply:
@CodingCat I asked @marcoabreu about this, he told me there is a difference between running on CPU and GPU. CPU usually runs multiple unit test at the same time and it may cause the problems on Spark. GPU runs individually in a single VM and that may not cause the same issues. 

However, we need to diagnose the GPU problem at least to get this PR merged.

Reply:
Just to clarify, in future we will run all tests (CPU and GPU) in parallel. At the moment it's only CPU ones which are already in parallel. Tests should not require exclusive access to any resource.